hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -307,22 +316,24 @@ public synchronized void handleTransaction(
 			Instant consensusTime,
 			com.swirlds.common.Transaction transaction
 	) {
-		if (isConsensus) {
-			ctx.logic().incorporateConsensusTxn(transaction, consensusTime, submittingMember);
+		try{
+			if (isConsensus) {
+				ctx.logic().incorporateConsensusTxn(transaction, txnAccessor, consensusTime, submittingMember);
+			}
+		} catch (InvalidProtocolBufferException exception) {
+			log.warn("Consensus platform txn was not gRPC!");
 		}
 	}
 
 	@Override
 	public void expandSignatures(Transaction platformTxn) {
 		try {
-			var accessor = new PlatformTxnAccessor(platformTxn);
+			setPlatformTxnAccessor(platformTxn);
 			expandIn(
-					accessor,
+					txnAccessor,
```

Comment:
The change here slightly changes original logic. Now line 333 may have `txnAccessor` as `null` if line 331`setPlatformTxnAccessor(platformTxn);` throws exception.  So it would be nice to add a test case for this scenario.

Reply:
Modified logic to use a separate component that would handle the conversion from `Transaction` to `PlatformTxnAccessor`.